.github/workflows/sonarcloud.yml

```
@@ -0,0 +1,53 @@
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  sonarcloud:
+    if: ${{ secrets.SONAR_TOKEN != '' }}
```

actionlint (1.7.11)

[error] context "secrets" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)